stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/internal/MessageListItemDiffCallback.kt

```diff
@@ -36,20 +41,25 @@ internal object MessageListItemDiffCallback : DiffUtil.ItemCallback<MessageListI
             )
             is MessageListItem.ThreadPlaceholderItem -> true
         }
+    }
 
     override fun getChangePayload(oldItem: MessageListItem, newItem: MessageListItem): Any? {
         return if (oldItem is MessageListItem.MessageItem) {
             newItem as MessageListItem.MessageItem
+            val oldMessage = oldItem.message
+            val newMessage = newItem.message
+
             MessageListItemPayloadDiff(
-                text = oldItem.message.text != newItem.message.text,
-                reactions = (oldItem.message.reactionCounts != newItem.message.reactionCounts) || (oldItem.message.reactionScores != newItem.message.reactionScores),
-                attachments = oldItem.message.attachments != newItem.message.attachments,
-                replies = oldItem.message.replyCount != newItem.message.replyCount,
-                syncStatus = oldItem.message.syncStatus != newItem.message.syncStatus,
-                deleted = oldItem.message.deletedAt != newItem.message.deletedAt,
+                text = oldMessage.text != newMessage.text,
+                reactions = (oldMessage.reactionCounts != newMessage.reactionCounts) || (oldMessage.reactionScores != newMessage.reactionScores),
+                attachments = oldMessage.attachments != newMessage.attachments,
+                replies = oldMessage.replyCount != newMessage.replyCount,
+                syncStatus = oldMessage.syncStatus != newMessage.syncStatus,
+                deleted = oldMessage.deletedAt != newMessage.deletedAt,
                 positions = oldItem.positions != newItem.positions,
-                pinned = oldItem.message.pinned != newItem.message.pinned,
-                user = oldItem.message.user != newItem.message.user,
+                pinned = oldMessage.pinned != newMessage.pinned,
+                user = oldMessage.user != newMessage.user,
+                mentions = (oldMessage.mentionedUsersIds != newMessage.mentionedUsersIds) || (oldMessage.mentionedUsers != newMessage.mentionedUsers)
```

Comment:
Might be worth double-checking but I'm almost sure we just need to check `oldMessage.mentionedUsers != newMessage.mentionedUsers`.
`mentionedUsersIds ` is being used only in `UpstreamMessageDto` and we are not receiving this field from the backend - we are receiving `mentionedUsers` instead

Reply:
That's a great suggestion. Indeed, the comparison base on `mentionedUsersIds` field is not needed. I tested and it works fine without it.